botocore/crt/auth.py

```diff
@@ -423,9 +438,12 @@ def _modify_request_before_signing(self, request):
         # parse_qs makes each value a list, but in our case we know we won't
         # have repeated keys so we know we have single element lists which we
         # can convert back to scalar values.
-        query_dict = dict(
-            [(k, v[0]) for k, v in
-             parse_qs(url_parts.query, keep_blank_values=True).items()])
+        query_dict = {
```

Comment:
This dict comprehension is a little wild, we had a similar one in the main `auth.py` too. Might be good to just clean these up.